src/backend/gram.rs

```diff
@@ -111,6 +173,13 @@ impl GramBackend {
             frontend::Expr::Lit(v) => Expr::Lit(v),
             frontend::Expr::Prop(e, props) => Expr::Prop(Box::new(self.convert_expr(*e)), props),
             frontend::Expr::Slot(s) => Expr::Slot(s),
+            frontend::Expr::List(es) => {
+                let mut items = Vec::new();
```

Comment:
I think you know the length of the list here so you can initialise the Vec with the desired capacity